### ISSUE TITLE
Parentheses in TSX/JSX markup result in unwanted line break

### ISSUE THREAD

**Issue by devuxer** [prettier/prettier#5275]
**Prettier 1.14.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeADgPlQZxgJ2gHNMAhOAQ3wEsojUB6PQuzAAgApgcBPAWwBGEADYA6AZRp0AvgAaAlIywgANCAjoY1aDmSgqhAO4AFKgl0oKANwjUAJqpAD8FMAGs4MAMrpXtIsgEAK5warQ4cPgwxi5EfBTIAGYUwhFqAFY4AB6kLu6eXhR8cAAytHBJKWkgvvgR+MhOFAI8wtCO6FIwAOr2MAAWyAAcAAxqnRAR3S7ojZ1w9VYVavhwAI5B1KsxFHEJSMmpoSARfNSB+CFqOP7CcACKQRDwlUdqMM29dgPIAEzvLmown8AGEIHx4o0oNBliAghEACrNCyHCLSaRAA)
```sh
--parser babylon
```

**Input:**
```jsx
<p><strong>Bearing</strong> ({symbol.bearing}°)</p>
```

**Output:**
```jsx
<p>
  <strong>Bearing</strong> ({symbol.bearing}
  °)
</p>;

```

**Expected behavior:**

```jsx
<p>
  <strong>Bearing</strong> ({symbol.bearing}°)
</p>;

```

<strike>A line break should not be added before the degree symbol, as this causes the user to see "30 °" instead of "30°". So, Prettier is not just changing the *formatting* of the code, it's changing the *meaning* of the code, which I would definitely consider a bug.</strike>

**Workaround:**

```jsx
<p><strong>Bearing</strong> {`(${symbol.bearing}°)`}</p>

```

**Comment by duailibe** [comment]
> A line break should not be added before the degree symbol, as this causes the user to see "30 °" instead of "30°".

This is not correct. A line break between two text nodes (i.e. `foo\nbar` will render `"foo bar"`) will create a space between them but not before/after a JSXTag or a JSXExpression, i.e. `{foo}\nbar` renders `"foobar"`.

See the transform in the Babel repl for confirmation: [link](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DwBwfAUABFwM4BcBOB7AdgczAIQKYEMkBLTYAekVUzCgAoBvOATwFsAjFAGwDo2DjMAX2hQADQEoI5cAG4gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=true&targets=&version=6.26.0&envVersion=)

In this case, it's only a matter of style and yeah, I guess we could make it *look* better.

**Comment by devuxer** [comment]
@duailibe, Ahh you're right. Seems to be the case for TypeScript as well. (I was basing my assertion off what happens if plain text is separated by a line break.)

**Comment by lydell** [comment]
Luckily, this has already been fixed by #5006! Stay tuned for the next release!

```
$ echo '<p><strong>Bearing</strong> ({symbol.bearing}°)</p>' | ./bin/prettier.js --parser babylon
<p>
  <strong>Bearing</strong> ({symbol.bearing}°)
</p>;
```

**Comment by lydell** [comment]
Duplicate of #4941
